<launch>
  <arg name="dt_design" default="10.0" description="[s]"/>
  <arg name="dx_design" default="30.0" description="[m]"/>
  <arg name="wz_threshold" default="0.01" description="[rad/s]"/>
  <arg name="vx_threshold" default="0.5" description="[m/s]"/>
  <arg name="predefined_coef_vx" default="1.00"/>
  <arg name="use_predefined_coef_vx" default="true"/>
  <arg name="results_path" default="/home/$(env USER)/output.txt"/>

  <arg name="estimation_freq" default="0.25" description="[Hz]"/>
  <arg name="show_debug_info" default="true"/>

  <arg name="use_pose_with_covariance" default="true"/>
  <arg name="input_pose_name" default="/localization/pose_estimator/pose"/>
  <arg name="input_pose_with_cov_name" default="/localization/pose_estimator/pose_with_covariance"/>

  <arg name="use_twist_with_covariance" default="true"/>
  <arg name="input_twist_name" default="/localization/twist_estimator/twist"/>
  <arg name="input_twist_with_cov_name" default="/localization/twist_estimator/twist_with_covariance"/>

  <arg name="output_bias_wz" default="estimated_bias_wz"/>
  <arg name="output_coef_vx" default="estimated_coef_vx"/>

  <node pkg="deviation_estimator" exec="deviation_estimator" name="deviation_estimator" output="screen">

    <remap from="in_pose" to="$(var input_pose_name)"/>
    <remap from="in_pose_with_covariance" to="$(var input_pose_with_cov_name)"/>

    <remap from="in_twist" to="$(var input_twist_name)"/>
    <remap from="in_twist_with_covariance" to="$(var input_twist_with_cov_name)"/>


    <param name="dt_design" value="$(var dt_design)"/>
    <param name="dx_design" value="$(var dx_design)"/>
    <param name="predefined_coef_vx" value="$(var predefined_coef_vx)"/>
    <param name="use_predefined_coef_vx" value="$(var use_predefined_coef_vx)"/>
    <param name="wz_threshold" value="$(var wz_threshold)"/>
    <param name="vx_threshold" value="$(var vx_threshold)"/>
    <param name="estimation_freq" value="$(var estimation_freq)"/>
    <param name="show_debug_info" value="$(var show_debug_info)"/>
    <param name="use_pose_with_covariance" value="$(var use_pose_with_covariance)"/>
    <param name="use_twist_with_covariance" value="$(var use_twist_with_covariance)"/>
    <param name="results_path" value="$(var results_path)"/>

    <remap from="estimated_bias_wz" to="$(var output_bias_wz)"/>
    <remap from="estimated_coef_vx" to="$(var output_coef_vx)"/>

  </node>

</launch>
